src/general/ActionHistory.cs

```diff
@@ -6,18 +6,18 @@
 ///   General implementation of an action history and undo / redo for use by editors
 /// </summary>
 /// <typeparam name="T">Type of actions to hold</typeparam>
-public class ActionHistory<T>
+public abstract class ActionHistory<T>
```

Comment:
I think instead of dirtying up the action history with sub actions and making this abstract, it'd be better to just make this a normal class, make things virtual, but leave the implementations otherwise as-is.

Reply:
I see your point, but why not make `ActionHistory` abstract? What does an instance of an `ActionHistory` supposed to be?

Reply:
It could be used for an text editor or another editor in the game. With just specifying a compatible action type.

Reply:
Actually now that I have had some time to think, why not just leave ActionHistory as is (as a general undo / redo stack) and make a CombinedAction class that allows you to shove multiple Action derived types in it. That way ActionHistory stays general, the actions stay simple, and the cell editor when needed can convert actions into CombinedActions and shove the "sub" steps into them. To me it doesn't make much sense that only partially undoing an action can work, it should be either all or nothing. Or in the rare case that is not true, the actions themselves that are allowed to fail should handle it internally.

Reply:
Seems like we both had the same idea, implemented that in the last commits